src/Order/OrderPaymentRefund.php

```diff
@@ -63,7 +64,7 @@ public function refund(UnitsRefunded $units): void
 
         $factoryName = $paymentMethod->getGatewayConfig()->getFactoryName() ?? null;
 
-        if (MollieGatewayFactory::FACTORY_NAME !== $factoryName) {
+        if (true === in_array($factoryName, [MollieGatewayFactory::FACTORY_NAME, MollieSubscriptionGatewayFactory::FACTORY_NAME], true)) {
```

Comment:
Can you put the logic into checker class and use it everywhere you do this in_array?